Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -889,22 +889,33 @@ private final class TokenStreamCreator: SyntaxVisitor {
   }
 
   override func visit(_ node: ClosureSignatureSyntax) -> SyntaxVisitorContinueKind {
-    let consistency: GroupBreakStyle
-    if let input = node.input, input.is(ClosureParamListSyntax.self) {
-      consistency = argumentListConsistency()
-    } else {
-      consistency = .inconsistent
-    }
-
     before(node.firstToken, tokens: .open)
 
     if let input = node.input {
       // We unconditionally put a break before the `in` keyword below, so we should only put a break
       // after the capture list's right bracket if there are arguments following it or we'll end up
       // with an extra space if the line doesn't wrap.
       after(node.capture?.rightSquare, tokens: .break(.same))
-      before(input.firstToken, tokens: .open(consistency))
-      after(input.lastToken, tokens: .close)
+
+      // When it's parenthesized, the input is a `ParameterClauseSyntax`. Otherwise, it's a
+      // `ClosureParamListSyntax`. The parenthesized version is wrapped in open/close breaks so that
+      // the parens create an extra level of indentation.
+      if let parameterClause = input.as(ParameterClauseSyntax.self) {
+        // Group outside of the parens, so that the entire argument list stays together if possible.
+        before(input.firstToken, tokens: .open)
+        after(input.lastToken, tokens: .close)
+
+        // Group and break inside of the parens to create a scope for the argument list, similar to
+        // a function call argument list.
+        let consistency = argumentListConsistency()
+        after(parameterClause.leftParen, tokens: .break(.open, size: 0), .open(consistency))
+        before(parameterClause.rightParen, tokens: .break(.close, size: 0), .close)
```

Comment:
I think these three lines can be replaced by a call to `arrangeParameterClause` with `forcesBreakBeforeRightParen: true`.

Reply:
Done.